addons/point_of_sale/static/src/app/store/models.js

```diff
@@ -555,14 +563,45 @@ export class Orderline extends PosModel {
     get_discount_str() {
         return this.discountStr;
     }
-    set_description(description) {
-        this.description = description || "";
-    }
     set_price_extra(price_extra) {
         this.price_extra = parseFloat(price_extra) || 0.0;
     }
-    set_full_product_name(full_product_name) {
-        this.full_product_name = full_product_name || "";
+    set_full_product_name() {
+        let attributeString = "";
+
+        if (this.attribute_value_ids && this.attribute_value_ids.length > 0) {
+            const filteredAttribute = Object.values(this.pos.db.attribute_by_id).sort(
+                (a, b) => a.sequence - b.sequence
+            );
+
+            for (const attribute of filteredAttribute) {
+                const valueIds = attribute.values
+                    .map((val) => val.id)
+                    .filter((id) => this.attribute_value_ids.includes(id));
+
+                for (const valId of valueIds) {
+                    const value = this.pos.db.attribute_value_by_id[valId];
+
+                    if (value.is_custom) {
+                        const customValue = this.custom_attribute_value_ids.find(
+                            (cus) =>
+                                cus.custom_product_template_attribute_value_id == parseInt(valId)
+                        );
+
+                        attributeString += customValue
+                            ? `${value.name}: ${customValue.custom_value}, `
+                            : `${value.name}, `;
+                    } else {
+                        attributeString += `${value.name}, `;
+                    }
+                }
+            }
+
+            attributeString = attributeString.slice(0, -2);
+            attributeString = `(${attributeString})`;
```

Comment:
```suggestion
            for (const valId of this.attribute_value_ids) {
                const value = this.pos.db.attribute_value_by_id[valId];
                if (value.is_custom) {
                    const customValue = this.custom_attribute_value_ids.find(
                        (cus) => cus.custom_product_template_attribute_value_id == parseInt(valId)
                    );
                    attributeString += customValue
                        ? `${value.name}: ${customValue.custom_value}, `
                        : `${value.name}, `;
                } else {
                    attributeString += `${value.name}, `;
                }
            }
            attributeString = attributeString.slice(0, -2);
            attributeString = `(${attributeString})`;
```

Can you check if this suggestion works? It's less nesting and I think it does the same thing.